CHANGELOG.md

```diff
@@ -4,10 +4,48 @@
 
 ### Added
 
+- `LtlDelay` delays a Ltl formula by n time stamps. `there` is now implemented
+  in terms of `LtlDelay`.
+- `viewByRef` and `previewByRef` which call `txSkelOutByRef` and apply a getter
+  and an affine fold on it, respectively.
+- Optics working on values in `Cooked.Skeleton.Output`
+- Missing instance of `IsTxSkelOutAllowedOwner` for `Versioned Script`
+- A new documentation file around optics, in `doc/OPTICS.md`
+
 ### Removed
 
+- Module `Cooked.Skeleton.Value`. `TxSkelOutValue` no longer exists and has been
+  replaced by two fields for `TxSkelOut`, `txSkelOutValue` and
+  `txSkelOutValueAutoAdjust`.
+- `unsafeTxOutByRef` which is now the default behavior for `txOutByRef`
+- Dedicated lookup functions for part of `TxSkelOut` such as
+  `typedDatumFromTxOutRef` as they have been replaced by `viewByRef` or
+  `previewByRef`.
+- `DatumContent`. In a `TxSkelOutDatum` the datum is now stored as is, and a new
+  type, `UtxoPayloadDatum` is used to store datum in utxo payloads.
+
 ### Changed
 
+- `TxOpts` replaced by `TxSkelOpts` and field names changed likewise.
+- `TxLabel` replaced by `TxSkelLabel`
+- `txOutByRef` replaced by `txSkelOutByRef`. It now throws an error if the
+  output is not found and thus it returns `m TxSkelOut` instead of `m (Maybe
+  TxSkelOut)`
+- Complete overhaul of the optics within skeleton. Their names has been made
+  consistent, the missing ones have been added and they are used more
+  consistently throughout the codebase.
+- `TxSkelMints` has been improved and clarified.
+- Transaction modifications in options are now called `txSkelModTx` and is now a
+  simple unwrapped function.
+- Parameters modifications in options are now called `txSkelModParams` and is
+  now a simple unwrapped function.
+- `ToCredential` is no longer a requirement for an owner of a `TxSkelOut`,
+  instead it is derived from `IsTxSkelOutAllowedOwner`
+- Some helpers, like `txSkelOutValue` are gone, as part of the optics overhaul,
+  and are replaced by their optics counterpart, such as `view txSkelOuValueL`. 
```

Comment:
At first sight, this worries me a little because the optics API is often more tedious to use.
But at the same time I like to prune some fat from the API.

I'm just wondering about consistency. Is it a new guideline for the project? From now on, if there are optics, we get rid of the legacy field accessors and helpers?

Reply:
I was wondering the same at first. Then I realized that by making everything an optic, then these helper would also be used as optics (and rarely as view XX) and thus that made sense.

Reply:
Yes, I think this should basically be a new guideline.